fftvis/beams.py

```diff
@@ -101,3 +101,98 @@ def get_apparent_flux_polarized(beam: np.ndarray, flux: np.ndarray):  # pragma:
         beam[0, 1, isrc] = i01 * flux[isrc]
         beam[1, 0, isrc] = np.conj(i01) * flux[isrc]
         beam[1, 1, isrc] = i11 * flux[isrc]
+
+@nb.jit(
+    nopython=True,
+    parallel=False,
+    nogil=False
+)
+def get_apparent_flux_polarized(beam: np.ndarray, coherency: np.ndarray): # pragma: no cover
+    """
+    Compute the matrix product A^H * C * A for each 2x2 matrix A in a stack.
+    
+    Parameters
+    ----------
+    beam : np.ndarray
+        An array of shape (2, 2, nstack) where each beam[:,:,i] is a 2x2 matrix A.
+    C : np.ndarray
+        A (2, 2, nstack) matrix.
+        
+    Returns
+    -------
+    out : np.ndarray
+        An array of shape (2, 2, nstack) where each slice is the result of A * C * A^H.
+    """
+    nstack = beam.shape[2]
+    
+    for i in range(nstack):
+        # Unpack elements of A from the i-th slice.
+        a00 = beam[0, 0, i]
+        a01 = beam[0, 1, i]
+        a10 = beam[1, 0, i]
+        a11 = beam[1, 1, i]
+        
+        # First multiplication: Compute tmp = A * C.
+        tmp00 = np.conj(a00) * coherency[0, 0, i] + np.conj(a01) * coherency[1, 0, i]
+        tmp01 = np.conj(a00) * coherency[0, 1, i] + np.conj(a10) * coherency[1, 1, i]
+        tmp10 = np.conj(a01) * coherency[0, 0, i] + np.conj(a11) * coherency[1, 0, i]
+        tmp11 = np.conj(a01) * coherency[0, 1, i] + np.conj(a11) * coherency[1, 1, i]
+        
+        # Second multiplication: Multiply tmp by A^H.
+        # A^H is the conjugate transpose of A.
+        beam[0, 0, i] = tmp00 * a00 + tmp01 * a10
+        beam[0, 1, i] = tmp00 * a01 + tmp01 * a11
+        beam[1, 0, i] = tmp10 * a00 + tmp11 * a10
+        beam[1, 1, i] = tmp10 * a01 + tmp11 * a11
+
+@nb.jit(
+    nopython=True,
+    parallel=False,
+    nogil=False
+)
+def get_apparent_flux_polarized_asymmetric_beam(beam1, coherency, beam2):
```

Comment:
If we have this, do we really need the non-asymmetric function above? Couldn't we just call this function with beam1 and beam2 pointing to the same array?